streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala

```diff
@@ -116,6 +121,33 @@ object KubernetesSessionClientV2 extends KubernetesClientV2Trait with Logger {
       ))
   }
 
+  @throws[Throwable]
+  def deploy(deployRequest: DeployRequest): DeployResponse = {
+
+    val richMsg: String => String = s"[flink-submit][appId=${deployRequest.id}] " + _
+
+    val flinkConfig = genFlinkConfig(deployRequest)
+    // Convert to FlinkDeployment CR definition
+    val flinkDeployDef = genFlinkDeployDef(deployRequest, flinkConfig) match {
+      case Right(result) => result
+      case Left(errMsg) =>
+        throw new IllegalArgumentException(
+          richMsg(s"Error occurred while parsing parameters: $errMsg"))
+    }
+
+    // Submit FlinkDeployment CR to Kubernetes
+    FlinkK8sOperator.deployApplicationJob(deployRequest.id, flinkDeployDef).runIOAsTry match {
```

Comment:
Please replace with `FlinkK8sOperator.deployCluster` api, `deployApplicationJob` api is for deploying flink application job